adr/0003-jsonpath-support.md

```diff
@@ -0,0 +1,43 @@
+# 3. JSONPath support
+
+Date: 15-03-2024
+
+## Status
+
+Draft
+
+## Context
+
+Support for JsonPATH was requested by some potential library clients.  
+However, JsonPATH RFC specifies way too many features which are tedious to implement without having runtime dependencies.  
+Therefore, we have to decide which features are necessary and which are not. 
+
+## Decisions
+
+The decisions are premised on the following expectations:
+1. Most of the clients will not use JsonPATH keys.  
+2. The correctness of the library takes precedent over the amount of features.
+
+Any implemented jsonpath feature adds some performance overhead both on configuration and on masking time, even if it is not used. In general, it also increases the likelihood of incorrect behaviour as the amount of code increases.
+Therefore, we decided to keep support for JsonPATH as limited as possible and to extend it only when someone specifically requests it.
+
+The following features are deemed necessary:
+1. Support for bracket, dot and mixed notations.  
+2. Support for object segments.  
+3. Support for array segments with wildcard indexing only.  
+4. Wildcard segments and selectors.   
+
+The rest of the features are deemed unnecessary.
```

Comment:
I would rather document the features we support according to [the RFC](https://www.rfc-editor.org/rfc/rfc9535.html), for example
1. [Name Selector](https://www.rfc-editor.org/rfc/rfc9535.html#name-name-selector) - Full support
2. [Wildcard Selector](https://www.rfc-editor.org/rfc/rfc9535.html#name-wildcard-selector) - Full support, except ambiguous segments
3. [Index Selector](https://www.rfc-editor.org/rfc/rfc9535.html#name-index-selector) - Not supported
4. [Array Slice Selector](https://www.rfc-editor.org/rfc/rfc9535.html#name-array-slice-selector) - Not supported
5. [Filter Selector](https://www.rfc-editor.org/rfc/rfc9535.html#name-filter-selector) - Not supported